### PR TITLE
Add 1 px safety offset in order to remove artifacts  

### LINKED ISSUE #2302: artifacts on canvas
The following control
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  
  <UserControl.Styles >
    <Style Selector="ItemsControl > ContentPresenter">
      <Setter Property="Canvas.Left" Value="{Binding Longitude}"/>
      <Setter Property="Canvas.Top" Value="{Binding Latitude}"/>
    </Style>
  </UserControl.Styles>
  <Panel>
    <Image Source="resm:SparkSDR.avalonia.Assets.0.png" Width="400" Height="400" />
  <ItemsControl Items="{Binding MapSpots}"  Width="400" Height="400">
    <ItemsControl.ItemsPanel>
      <ItemsPanelTemplate>
        <Canvas/>               
      </ItemsPanelTemplate>
    </ItemsControl.ItemsPanel>
    <ItemsControl.DataTemplates>
      <DataTemplate>
         <Ellipse Fill="{Binding brush}" Width="5" Height="5"  />
       </DataTemplate>
    </ItemsControl.DataTemplates>
  </ItemsControl>
 </Panel>
 
</UserControl>
```
when left running for a long time with MapSpots being updated it left behind traces of old ellipses
mapartifacts
As it is the effect is quite good in my app :)

### LINKED ISSUE #9659: Custom render control leaves "spots" in background
**Describe the bug**
I created a custom render templated control which represents the Windows 10 busy indicator with circulating dots. This uses ```context.DrawEllipse```. After each circle of the dots a spot in the color of the dots is left in the background. This spot becomes more and more visible after each circle iteration. Hence, it seems that the background is not correctly cleared. This is visible differently on different machines. Sometimes its in the top row, sometimes its left, sometimes it is both. NOTE: the problem does not appear in the designer view within Visual Studio

**To Reproduce**
Use the busy indicator and set to active.

**Expected behavior**
The "spots" should not appear.

**Screenshots**
I have created screenshots during active animation. I have set the background to darg gray to visualize that the spot happens within the controls view port.
image

Note: the spots appear outside the dot drawing area which is set to view port - 1.
image

**Desktop (please complete the following information):**

- OS: Windows 10
- Version: 11.0.0-preview4

**Additional context**
This is the custom render method:
```csharp
        public override void Render(DrawingContext context)
        {
            if (!IsActive || DotCount == 0) { return; }

            double size = Math.Min(Bounds.Width, Bounds.Height);
            double dotRadius = 0.05 * size;
            Circle2D circle = new Circle2D(0.5 * Bounds.Width, 0.5 * Bounds.Height, 0.5 * size - dotRadius - 1);

            Interval range = new Interval(Phase, Phase + 0.06 * (DotCount - 1));

            foreach (double t in range.Discretize(DotCount))
            {
                Point2D p = circle.GetPointForAngle(GetAngleAt(t));
                context.DrawEllipse(Brush, null, new Point(p.X, p.Y), dotRadius, dotRadius);
            }
        }
```

You can find the full code of the busy indicator here: https://github.com/llfab/Samples/tree/main/BusyIndicator

### PR DESCRIPTION
**Pull request by timunie**
## What does the pull request do?
Some geometries are not rendered correctly and some artifacts are left, see https://github.com/AvaloniaUI/Avalonia/issues/9659

## What is the current behavior?
Artifacts are there in some cases

## What is the updated/expected behavior with this PR?
No artifacts

## How was the solution implemented (if it's not obvious)?
Add 1 pixel of safety area to dirty rects

## Checklist

- ~[ ] Added unit tests (if possible)?~
- ~[ ] Added XML documentation to any related classes?~
- ~[ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation~

## Fixed issues
Fixes #9659 (to be confirmed)
Fixes #2302 (to be confirmed)

This issue is marked as draft as the issues should be confirmed first

### CODE REVIEW

File: src/Avalonia.Base/Rendering/SceneGraph/DrawOperation.cs [hunk 1]
```diff
@@ -11,7 +11,10 @@ internal abstract class DrawOperation : IDrawOperation
     {
         public DrawOperation(Rect bounds, Matrix transform)
         {
-            bounds = bounds.Normalize().TransformToAABB(transform);
+            bounds = bounds
```
Comment (kekekeks) [inline src/Avalonia.Base/Rendering/SceneGraph/DrawOperation.cs:14]: I'm not quite sure it's a good idea to inflate every single draw operation, especially plain rects that are used as backgrounds. That might cause **way more** elements to be drawn in a particular frame.

Comment (robloo) [reply src/Avalonia.Base/Rendering/SceneGraph/DrawOperation.cs:14]: I agree, this whole PR is unfortunately a hack and probably isn't a good idea. It's better to find and fix the rounding errors.

Comment (timunie) [reply src/Avalonia.Base/Rendering/SceneGraph/DrawOperation.cs:14]: I somewhat agree with both of you. But I want to know if that fix would at least solve the issue. Maybe only some geometries needs this fix. After the PR is identified as a valid solution, the cases where this is needed can be refined.

**Comment by timunie** [comment]
Closing this as it was only a workaround and after testing, it didn't even solve the reported issue.

**Pull request closed without merging**